imap_processing/swe/l1b/swe_l1b.py

```diff
@@ -1,48 +1,928 @@
-        The l1a data input.
+    esa_table_number : int
+        ESA table index number.
 
     Returns
     -------
-     data : xarray.Dataset
-        Processed data to L1B.
+    esa_steps : pandas.DataFrame
+        ESA table_number and its associated values.
     """
-    apid = int(l1a_dataset.attrs["packet_apid"])
+    if esa_table_number not in [0, 1]:
+        raise ValueError(f"Unknown ESA table number {esa_table_number}")
 
-    # convert value from raw to engineering units as needed
-    conversion_table_path = str(
-        imap_module_directory / "swe/l1b/engineering_unit_convert_table.csv"
+    # Get the lookup table DataFrame
+    lookup_table = read_lookup_table()
+
+    esa_steps = lookup_table.loc[lookup_table["table_index"] == esa_table_number]
+    return esa_steps
+
+
+def deadtime_correction(
+    counts: np.ndarray, acq_duration: Union[int, npt.NDArray]
+) -> npt.NDArray:
+    """
+    Calculate deadtime correction.
+
+    Deadtime correction is a technique used in various fields, including
+    nuclear physics, radiation detection, and particle counting, to compensate
+    for the effects of the time period during which a detector is not able to
+    record new events or measurements after detecting a previous event.
+    This "deadtime" is essentially the time during which the detector is
+    recovering from the previous detection and is unable to detect new events.
+
+    In particle detectors, there is a finite time required for the detector to
+    reset or recover after detecting a particle. During this deadtime, any
+    subsequent particles that may have arrived go undetected. As a result,
+    the recorded count rate appears to be lower than the actual count rate.
+
+    Deadtime correction involves mathematically adjusting the measured count
+    rates to compensate for this deadtime effect. This correction is crucial
+    when dealing with high-intensity sources or particle fluxes, as the deadtime
+    can significantly affect the accuracy of the measurements.
+
+    Deadtime correction is important to ensure accurate measurements and data
+    analysis in fields where event detection rates are high and where every
+    detected event is critical for understanding physical processes.
+
+    Parameters
+    ----------
+    counts : numpy.ndarray
+        Counts data before deadtime corrections.
+    acq_duration : int or numpy.ndarray
+        This is ACQ_DURATION from science packet. acq_duration is in microseconds.
+
+    Returns
+    -------
+    corrected_count : numpy.ndarray
+        Corrected counts.
+    """
+    # deadtime is 360 ns
+    deadtime = 360e-9
+    if isinstance(acq_duration, int):
+        # Convert acq_duration to a numpy array for consistency
+        acq_duration = np.array([acq_duration])
+    correct = 1.0 - (deadtime * (counts / (acq_duration[..., np.newaxis] * 1e-6)))
+    correct = np.maximum(0.1, correct)
+    corrected_count = np.divide(counts, correct)
```

Comment:
I think you might want to bring the below astype() up into this to force counts to be float64 before the division? Same for the below function.
```suggestion
    corrected_count = counts.astype(np.float64) / correct
```